app/controllers/application_controller.rb

```diff
@@ -0,0 +1,21 @@
+class ApplicationController < ActionController::API
+  rescue_from ActiveRecord::RecordInvalid, with: :handle_record_invalid
+  rescue_from ActionController::ParameterMissing, with: :handle_parameter_missing
```

Comment:
where are these methods being used?

Reply:
I think that they are invoked automatically whenever these exceptions are raised within any controller that inherits from ApplicationController